### PR TITLE
Tricrypto `remove_liquidity_one_coin` and `remove_liquidity`

### LINKED ISSUE #218: Tricrypto `remove_liquidity_one_coin`
Currently `CurveCryptoPool.remove_liquidity_one_coin` does not handle 3 coins, only 2. Update the implementation in a clean manner to handle this. Similar test as for 2 coins should work. Reference implementation is the Tricrypto-NG vyper fixture.

**Note:** This is blocked by `calc_withdraw_one_coin`.

### PR DESCRIPTION
**Pull request by allt0ld**
### Description

Closes #218.

- Made `CurveCryptoPool`'s `remove_liquidity` and `remove_liquidity_one_coin` functions work for 3-coin pools.
- New tests `test_remove_liquidity` and `test_remove_liquidity_one_coin` in `test_tricrypto.py` are borrowed directly from `test_cryptopool.py`.

### Hygiene checklist

- [x] Changelog entry
- [x] Everything public has a Numpy-style docstring
      (modules, public functions, classes, and public methods)
- [x] Commit history is cleaned-up with minor changes squashed together
      and descriptive commit messages following [Tim Pope's style](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: test/unit/test_tricrypto.py [hunk 1]
```diff
@@ -560,6 +560,54 @@ def test_price_oracle(vyper_tricrypto, price_oracle, last_prices, time_delta):
...
+    expected_D = vyper_tricrypto.D()
+
+    pool.remove_liquidity(amount)
+
+    assert pool.balances == expected_balances
+    assert pool.tokens == expected_lp_supply
+    assert pool.D == expected_D
+
+
+@given(lp_tokens, st.integers(min_value=0, max_value=1))
```
Comment (nagakingg) [inline test/unit/test_tricrypto.py:587]: Unlikely to matter, but max_value should be 2, right?

File: curvesim/pool/cryptoswap/pool.py [hunk 2]
```diff
@@ -840,23 +840,38 @@ def remove_liquidity(
         ----
         "This withdrawal method is very safe, does no complex math"
         """
-        min_amounts = min_amounts or [0, 0]
+        n_coins: int = self.n
+        min_amounts = min_amounts or [0] * n_coins
+
+        amount: int = _amount
+        balances: List[int] = self.balances
+        d_balances: List[int] = [0] * n_coins
```
Comment (chanhosuh) [inline curvesim/pool/cryptoswap/pool.py:848]: i would call this something more readable like `withdraw_amounts`.

File: curvesim/pool/cryptoswap/pool.py [hunk 3]
```diff
@@ -878,18 +893,20 @@ def remove_liquidity_one_coin(
...
-        xp = [0] * self.n
+        xp: List[int] = [0] * self.n
 
         self._claim_admin_fees()
 
+        calc_price: bool = self.n == 2
+
         dy, p, D, xp = self._calc_withdraw_one_coin(
-            A, gamma, token_amount, i, False, True
+            A, gamma, token_amount, i, False, calc_price
```
Comment (chanhosuh) [inline curvesim/pool/cryptoswap/pool.py:909]: nice!

File: test/unit/test_tricrypto.py [hunk 4]
```diff
@@ -560,6 +560,54 @@ def test_price_oracle(vyper_tricrypto, price_oracle, last_prices, time_delta):
...
 
+@given(lp_tokens)
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    max_examples=5,
+    deadline=None,
+)
+def test_remove_liquidity(vyper_tricrypto, amount):
+    """Test `remove_liquidity` against vyper implementation."""
+    assume(amount < vyper_tricrypto.totalSupply())
```
Comment (chanhosuh) [inline test/unit/test_tricrypto.py:571]: let's test the edge case of when amount is total supply

File: test/unit/test_tricrypto.py [hunk 5]
```diff
@@ -560,6 +560,54 @@ def test_price_oracle(vyper_tricrypto, price_oracle, last_prices, time_delta):
...
+    deadline=None,
+)
+def test_remove_liquidity_one_coin(vyper_tricrypto, amount, i):
+    """Test `remove_liquidity_one_coin` against vyper implementation."""
+    assume(amount < vyper_tricrypto.totalSupply())
+
+    pool = initialize_pool(vyper_tricrypto)
+
+    vyper_tricrypto.remove_liquidity_one_coin(amount, i, 0)
+    expected_coin_balance = vyper_tricrypto.balances(i)
```
Comment (chanhosuh) [inline test/unit/test_tricrypto.py:600]: we might as well check all the balances to make sure something weird didn't happen.

**Review by chanhosuh — CHANGES_REQUESTED** [review]

**Review by chanhosuh — APPROVED** [review]
